tests/integration/test_connection.py

```diff
@@ -147,12 +147,6 @@ def __init__(self, destination):
         if hasattr(ssl, 'PROTOCOL_TLS_SERVER'):
```

Comment:
Will this `if` ever fail? Do we really want to do nothing if it is false? This seems like a potential way in for a silent failure